lib/matplotlib/scale.py

```
@@ -265,6 +266,112 @@ def set_default_locators_and_formatters(self, axis):
             axis.set_minor_locator(NullLocator())
 
 
+class PowerTransform(Transform):
+    """
+    A simple power transformation used by `.PowerScale`.
+
+    This transformation applies a power-law scaling to positive values, while
+    nonpositive values remain unchanged.
+    """
+    input_dims = output_dims = 1
+
+    def __init__(self, gamma):
+        """
+        Parameters
+        ----------
+        gamma : float
+            Power law exponent.
+        """
+        super().__init__()
+        self.gamma = gamma
+
+    def __str__(self):
+        return "{}(gamma={})".format(
+            type(self).__name__, self.gamma)
+
+    def transform_non_affine(self, a):
+        with np.errstate(divide="ignore", invalid="ignore"):
+            mask = np.ma.getmask(a)
+            d = np.asarray(a.data)
+            out = np.where(d > 0, np.power(d, self.gamma), d)
+            mout = np.ma.masked_array(out, mask=mask)
+            return mout
+
+    def inverted(self):
+        return InvertedPowerTransform(self.gamma)
+
+
+class InvertedPowerTransform(Transform):
+    """
+    The inverse of the `.PowerTransform`.
+
+    This transformation applies an inverse power-law scaling to positive values,
+    while nonpositive values remain unchanged.
+    """
+    input_dims = output_dims = 1
+
+    def __init__(self, gamma):
+        """
+        Parameters
+        ----------
+        gamma : float
+            Power law exponent.
+        """
+        super().__init__()
+        self.gamma = gamma
+
+    def transform_non_affine(self, a):
+        if self.gamma == 0:
+            return np.inf
```

@Sreekanth-M8 Why do we have this? Is this previously expected behavior? If not, we need to decide whether

1. We should raise an error
2. Return an **array** of np.inf of same shape as `a`.

---

Sorry I thought of creating an array.

No, this is not previously expected behavior, in previous code they didn't implement any check for `gamma == 0`.

I think it's better to raise an error as we can't say for sure if the inverse is infinite.

---

Should we also check for negative values of gamma?

---

I think we should move this check itself to the `__init__` function. See what LogTransform does for example. If base is between 0 and 1, it raises a ValueError. Similarly, other transforms also have this check within the `__init__` function.